File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        vector<vector<char>> freq(s.size());
        unordered_map<char, int> count; 
        string res = "";
        for (char c: s) {
            count[c]++;
        }
        for (auto &p: count) {
            freq[p.second - 1].push_back(p.first);
        }
        for (int i = s.size() - 1; i >= 0; i--) {
            for (char c: freq[i]) {
                for (int j = 0; j < i + 1; j++) {
                    res += c;
                }
            }
        }
        return res;
    }
};

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        count = {}
        for c in s: 
            count[c] = count.get(c, 0) + 1
            
        freq = [[] for i in range(len(s))]
        for c in count:
            freq[count[c] - 1].append(c)
            
        res = ""
        for i in range(len(s) - 1, -1, -1):
            for c in freq[i]:
                res += c * (i + 1)
                
        return res;